tests/net/socket/tls/src/main.c

```diff
@@ -0,0 +1,475 @@
+ *   client authentication is required)
+ */
+enum tls_tag {
+	/** The Certificate Authority public key */
+	CA_CERTIFICATE_TAG,
+	/** Used for both the public and private server keys */
+	SERVER_CERTIFICATE_TAG,
+	/** Used for both the public and private client keys */
+	CLIENT_CERTIFICATE_TAG,
+};
+
+/** @brief synchronization object for server & client threads */
+static struct k_sem server_sem;
+
+/** @brief The server thread stack */
+static K_THREAD_STACK_DEFINE(server_stack, STACK_SIZE);
+/** @brief the server thread object */
+static struct k_thread server_thread;
+
+#ifdef CONFIG_TLS_CREDENTIALS
+/**
+ * @brief The Certificate Authority (CA) Certificate
+ *
+ * The client needs the CA cert to verify the server public key. TLS client
+ * sockets are always required to verify the server public key.
+ *
+ * Additionally, when the peer verification mode is
+ * @ref TLS_PEER_VERIFY_OPTIONAL or @ref TLS_PEER_VERIFY_REQUIRED, then
+ * the server also needs the CA cert in order to verify the client. This
+ * type of configuration is often referred to as *mutual authentication*.
+ */
+static const unsigned char ca[] = {
+#include "ca.inc"
+};
+
+/**
+ * @brief The Server Certificate
+ *
+ * This is the public key of the server.
+ */
+static const unsigned char server[] = {
+#include "server.inc"
+};
+
+/**
+ * @brief The Server Private Key
+ *
+ * This is the private key of the server.
+ */
+static const unsigned char server_privkey[] = {
+#include "server_privkey.inc"
+};
+
+/**
+ * @brief The Client Certificate
+ *
+ * This is the public key of the client.
+ */
+static const unsigned char client[] = {
+#include "client.inc"
+};
+
+/**
+ * @brief The Client Private Key
+ *
+ * This is the private key of the client.
+ */
+static const unsigned char client_privkey[] = {
+#include "client_privkey.inc"
+};
+#else /* CONFIG_TLS_CREDENTIALS */
+#define ca NULL
+#define server NULL
+#define server_privkey NULL
+#define client NULL
+#define client_privkey NULL
+#endif /* CONFIG_TLS_CREDENTIALS */
+
+/**
+ * @brief The server thread function
+ *
+ * This function simply accepts a client connection and
+ * echoes the first @ref SECRET_SIZE bytes of the first
+ * packet. After that, the server is closed and connections
+ * are no longer accepted.
+ *
+ * @param arg0 a pointer to the int representing the server filed descriptor
+ * @param arg1 ignored
+ * @param arg2 ignored
+ */
+static void server_thread_fn(void *arg0, void *arg1, void *arg2)
+{
+	const int server_fd = POINTER_TO_INT(arg0);
+
+	int r;
+	int client_fd;
+	socklen_t addrlen;
+	char addrstr[INET_ADDRSTRLEN];
+	struct sockaddr_in sa;
+	char *addrstrp;
+
+	k_thread_name_set(k_current_get(), "server");
+
+	NET_DBG("Server thread running");
+
+	memset(&sa, 0, sizeof(sa));
+	addrlen = sizeof(sa);
+
+	NET_DBG("Accepting client connection..");
+	k_sem_give(&server_sem);
+	r = accept(server_fd, (struct sockaddr *)&sa, &addrlen);
+	__ASSERT(r != -1, "accept() failed (%d)", r);
```

Comment:
You should use `zassert*()` macros like in other functions in this file.

Reply:
Does that work properly when called from a separate thread? I vaguely remember that the test did not fail reliably when I used `zassert()` here.

Reply:
Works fine, my mistake.